frank-doc-doclet/src/main/java/org/frankframework/frankdoc/feature/AbstractNonValuedFeature.java

```diff
@@ -0,0 +1,47 @@
+/* 
+Copyright 2022 WeAreFrank! 
+
+Licensed under the Apache License, Version 2.0 (the "License"); 
+you may not use this file except in compliance with the License. 
+You may obtain a copy of the License at 
+
+    http://www.apache.org/licenses/LICENSE-2.0 
+
+Unless required by applicable law or agreed to in writing, software 
+distributed under the License is distributed on an "AS IS" BASIS, 
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+See the License for the specific language governing permissions and 
+limitations under the License. 
+*/
+
+package org.frankframework.frankdoc.feature;
+
+import org.frankframework.frankdoc.wrapper.FrankClass;
+import org.frankframework.frankdoc.wrapper.FrankDocException;
+import org.frankframework.frankdoc.wrapper.FrankEnumConstant;
+import org.frankframework.frankdoc.wrapper.FrankMethod;
+
+abstract class AbstractNonValuedFeature {
+	private final String javaAnnotation;
+	private final String javaDocTag;
+
+	AbstractNonValuedFeature(String javaAnnotation, String javaDocTag) {
+		this.javaAnnotation = javaAnnotation;
+		this.javaDocTag = javaDocTag;
+	}
```

Comment:
Waarom heb je er niet voor gekozen om dit bij het Feature enum te betrekken? Het hoort immers allemaal bij elkaar.

Je zou daar iets van een ValueResolver kunnen gebruiken. De ValueResolver zou een FunctionalInterface kunnen zijn, dan kan je of een Klasse er voor maken, voor de lastigere implementaties zoals `Default` of een lambda kunnen meegeven voor de makkelijkere zoals `Optional`.

Dan scheelt het code, en betrek je de implementaties om aan een value te komen bij het Feature.

Reply:
Ik snap nog niet helemaal hoe dit code kan schelen. Voor de ingewikkelde gevallen Description, Reference en Default heb je een aparte klasse nodig. Ik zag niet goed hoe ik alles bij elkaar kon houden in de enum, dus heb ik de Feature enum die we hadden weggehaald.

Er is ook een verschil tussen features die een boolean karakter hebben (wel of niet aanwezig) en features waar een waarde bij hoort. Bij het ene soort feature hoef ik nu geen dummy methods voor het andere te hebben. De interfaces van ieder van de feature klasses is nu duidelijker omdat ze van elkaar mogen verschillen.